From: ramana <ramana@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 11 Nov 2013 09:38:14 +0000
Subject: [PATCH] Backport fix for PR target/58854

2013-11-11  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>

       Backported from mainline
        2013-10-30  Ramana Radhakrishnan  <ramana.radhakrishnan@arm.com>

       PR target/58854
       * config/arm/arm.c (arm_expand_epilogue_apcs_frame): Emit blockage



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@204665 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/config/arm/arm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gcc/config/arm/arm.c b/gcc/config/arm/arm.c
index d787a6e..52871f1 100644
--- a/gcc/config/arm/arm.c
+++ b/gcc/config/arm/arm.c
@@ -23555,6 +23555,7 @@ arm_expand_epilogue_apcs_frame (bool really_return)
   num_regs = bit_count (saved_regs_mask);
   if ((offsets->outgoing_args != (1 + num_regs)) || cfun->calls_alloca)
     {
+      emit_insn (gen_blockage ());
       /* Unwind the stack to just below the saved registers.  */
       emit_insn (gen_addsi3 (stack_pointer_rtx,
                              hard_frame_pointer_rtx,
